gapis/api/vulkan/frame_loop.go

```diff
@@ -105,14 +104,28 @@ type frameLoop struct {
 	loopCountPtr value.Pointer
 
 	frameNum uint32
+
+	loopTerminated      bool
+	lastObservedCommand api.CmdID
 }
 
-func newFrameLoop(ctx context.Context, c *capture.GraphicsCapture, numInitialCmds int, Cmds []api.Cmd, loopCount int32) *frameLoop {
-	f := &frameLoop{
-		capture:        c,
-		cmds:           Cmds,
-		numInitialCmds: numInitialCmds,
-		loopCount:      loopCount,
+func newFrameLoop(ctx context.Context, graphicsCapture *capture.GraphicsCapture, loopStart api.CmdID, loopEnd api.CmdID, loopCount int32) *frameLoop {
+
+	if api.CmdID.Real(loopStart) >= api.CmdID.Real(loopEnd) {
+		log.F(ctx, true, "FrameLoop: Cannot create FrameLoop for zero or negative length loop")
+	}
```

Comment:
You should return nil, or even more specifically, have this return (*frameLoop, error)

Reply:
log.F doesn't cause a panic?

Reply:
Oh, I think you are right, F is fatal, sorry.